os/net/mac/tsch/tsch-schedule.c

```diff
@@ -226,9 +226,11 @@ tsch_schedule_add_link(struct tsch_slotframe *slotframe,
       return NULL;
     }
 
-    /* Start with removing the link currently installed at this timeslot (needed
-     * to keep neighbor state in sync with link options etc.) */
-    tsch_schedule_remove_link_by_timeslot(slotframe, timeslot, channel_offset);
+    if(do_remove) {
+      /* Start with removing the link currently installed at this timeslot (needed
+       * to keep neighbor state in sync with link options etc.) */
+      tsch_schedule_remove_link_by_timeslot(slotframe, timeslot, channel_offset);
+    }
```

Comment:
Does having multiple slots at the same coordinates have any implications in the scheduler? I feel like not, as it's not much different to handle as having multiple choffsets at the same timeslot. Just something to think about.

Reply:
Ok, there is actually an issue. 

The active cell selection will be working fine, at least not worse than previously. The PR #1162 solves the case when there are multiple Tx links at the same timeslot. In general, we still don't have a good solution for multiple Rx links - just the first one always is used. Round robin would be better, but adds some complexity.

But the real issue is in the unlikely but possible case when the Orchestra code has added Tx and Rx links at the same (timeslot, choffset) and wants to remove one of them. The wrong link may be removed, since we don't look at link options when removing links. Not sure what's the best way to fix this: adding a function that searches for links including their options would be one way.

Reply:
I modified `tsch_schedule_remove_link_by_timeslot` to remove all matching links.

This will have no direct effect on the new rule, since it does no call the function `tsch_schedule_remove_link_by_timeslot` after the changes.

Instead, there is a new static function `remove_unicast_link` in the file `orchestra-rule-unicast-link-based.c` that also looks at link's options to decide whether that link should be removed. If it is, the code calls `tsch_schedule_remove_link` with the link to remove as an argument.